pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -102,7 +102,7 @@
 
     private final ConsumerStatsImpl stats;
 
-    private volatile int maxUnackedMessages;
+    private final boolean maxUnackedMessagesEnabled;
```

Comment:
If we update Policies, it should be possible to disable this feature. If it is set to final, how to update it later?

Reply:
This `maxUnackedMessagesEnabled` is used for the purpose of disabling "unacked messages check" feature on consumer level for  non-durable subscriptions. Following previous logic, see

https://github.com/apache/pulsar/blob/af761e2f8d3aee35c2cf081998f556972a92b574/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L756-L758